pkg/collector/collector.go

```diff
@@ -89,66 +87,100 @@ func (collector *Collector) Collect(ch chan<- prometheus.Metric) {
+		log.Debug("memUsed: ", float(matches[3][1])*1024)
+		ch <- prometheus.MustNewConstMetric(collector.poolsUsed, prometheus.GaugeValue, float(matches[6][1])*1024)
+		log.Debug("poolsUsed: ", float(matches[6][1])*1024)
+		ch <- prometheus.MustNewConstMetric(collector.poolsTotal, prometheus.GaugeValue, float(matches[7][1])*1024)
+		log.Debug("poolsTotal: ", float(matches[7][1])*1024)
 	}
+}
 
-	if len(matches) > 0 && idle == true {
+func (collector *Collector) CollectThreads(ch chan<- prometheus.Metric, stats string) {
+	regex := regexp.MustCompile(`(?:THREADS:\slive|idle|max|idle-timeout)\s+([0-9.]+|N\/A)+`)
+	matches := regex.FindAllStringSubmatch(stats, -1)
+
+	log.Debug("Matches Threads", matches)
+
+	//THREADS
+	if len(matches) > 0 {
 		ch <- prometheus.MustNewConstMetric(collector.threadsLive, prometheus.GaugeValue, float(matches[1][1]))
+		log.Debug("threadsLive: ", float(matches[1][1]))
 		ch <- prometheus.MustNewConstMetric(collector.threadsIdle, prometheus.GaugeValue, float(matches[2][1]))
+		log.Debug("threadsIdle: ", float(matches[2][1]))
 		ch <- prometheus.MustNewConstMetric(collector.threadsMax, prometheus.GaugeValue, float(matches[3][1]))
-		ch <- prometheus.MustNewConstMetric(collector.queue, prometheus.GaugeValue, float(matches[5][1]))
-		ch <- prometheus.MustNewConstMetric(collector.memHeap, prometheus.GaugeValue, float(matches[8][1])*1024)
-		ch <- prometheus.MustNewConstMetric(collector.memMmap, prometheus.GaugeValue, float(matches[9][1])*1024)
-		ch <- prometheus.MustNewConstMetric(collector.memUsed, prometheus.GaugeValue, float(matches[10][1])*1024)
-		ch <- prometheus.MustNewConstMetric(collector.poolsUsed, prometheus.GaugeValue, float(matches[14][1])*1024)
-		ch <- prometheus.MustNewConstMetric(collector.poolsTotal, prometheus.GaugeValue, float(matches[15][1])*1024)
+		log.Debug("threadsMax: ", float(matches[3][1]))
 	}
+}
 
-	version := collector.client.Dial(commands.VERSION)
-	regex = regexp.MustCompile(`((ClamAV)+\s([0-9.]*)/([0-9.]*))`)
-	matches = regex.FindAllStringSubmatch(string(version), -1)
+func (collector *Collector) CollectQueue(ch chan<- prometheus.Metric, stats string) {
+	regex := regexp.MustCompile(`(?:QUEUE:|FILDES|STATS)\s+([0-9.]+|N\/A)`)
+	matches := regex.FindAllStringSubmatch(stats, -1)
+
+	log.Debug("Matches Queue", matches)
+
+	//QUEUE
 	if len(matches) > 0 {
-		ch <- prometheus.MustNewConstMetric(collector.buildInfo, prometheus.GaugeValue, 1, matches[0][3], matches[0][4])
+		ch <- prometheus.MustNewConstMetric(collector.queue, prometheus.GaugeValue, float(matches[0][1]))
+		log.Debug("queue: ", float(matches[0][1]))
 	}
+}
+
+func (collector *Collector) CollectBuildInfo(ch chan<- prometheus.Metric) {
+	version := collector.client.Dial(commands.VERSION)
+	regex := regexp.MustCompile(`ClamAV\s([0-9.]*)/(\d+)/(.+)`)
+	matches := regex.FindStringSubmatch(string(version))
+
+	log.Debug("Matches Version", matches)
```

Comment:
Could you add a `--debug` flag and set the log level in the logger accordingly when setting the flag?

Reply:
Done